### PR TITLE
Added implementation

### PR DESCRIPTION
**Pull request by oleksandrdziubiuk**

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 1]
```diff
@@ -56,4 +56,47 @@ public MovieSession add(MovieSession session) {
...
+
+    @Override
+    public void delete(Long id) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.delete(id);
+            transaction.commit();
```
Comment (engineer027) [inline src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java]:
```suggestion
            transaction = session.beginTransaction();
            MovieSession movieSession = session.load(MovieSession.class, id);
            session.delete(movieSession);
            transaction.commit();
```

File: src/main/java/com/dev/cinema/model/dto/MovieSessionResponseDto.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package com.dev.cinema.model.dto;
+
+public class MovieSessionResponseDto {
+    private String showTime;
+    private Long movieId;
+    private Long cinemaHallId;
```
Comment (engineer027) [inline src/main/java/com/dev/cinema/model/dto/MovieSessionResponseDto.java:8]:
```suggestion
    private String showTime;
    private Long movieId;
    private Long cinemaHallId;
    private Long id;
    private String movieTitle;
```

File: src/main/java/com/dev/cinema/service/dtomapper/MovieSessionMapper.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        return responseDto;
+    }
+
+    public MovieSession fromDto(MovieSessionRequestDto requestDto) {
+        MovieSession movieSession = new MovieSession();
+        movieSession.setMovie(movieService.get(requestDto.getMovieId()));
+        LocalDateTime localDate = LocalDateTime.parse(requestDto.getShowTime(),
+                DateTimeFormatter.ISO_LOCAL_DATE_TIME);
+        movieSession.setShowTime(localDate);
+        movieSession.setCinemaHall(cinemaHallService.get(requestDto.getCinemaHallId()));
```
Comment (engineer027) [inline src/main/java/com/dev/cinema/service/dtomapper/MovieSessionMapper.java:39]: set id and movieTitle

**Review by engineer027 — APPROVED** [review]

**Review by engineer027 — APPROVED** [review]

File: src/main/java/com/dev/cinema/controller/MovieSessionController.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    private final MovieSessionMapper movieSessionMapper;
+
+    public MovieSessionController(MovieSessionService movieSessionService,
+                                  MovieSessionMapper movieSessionMapper) {
+        this.movieSessionService = movieSessionService;
+        this.movieSessionMapper = movieSessionMapper;
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAvailableSession(@RequestParam Long id,
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controller/MovieSessionController.java]:
```suggestion
    public List<MovieSessionResponseDto> getAvailableSession(@RequestParam Long movieId,
```

File: src/main/java/com/dev/cinema/model/dto/MovieSessionResponseDto.java (hunk 2 above)
Comment (KseniiaMakarova) [reply src/main/java/com/dev/cinema/model/dto/MovieSessionResponseDto.java:8]: @engineer027 previously it seems to be ok as well
the point is that we should return sessionId, showTime as String and some info about Movie and CinemaHall - id or other information

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**